src/hid_unix.c

```diff
@@ -66,9 +66,16 @@ fido_hid_unix_wait(int fd, int ms, const fido_sigset_t *sigmask)
 		ts.tv_nsec = (ms % 1000) * 1000000;
 	}
 
-	if ((r = ppoll(&pfd, 1, ms > -1 ? &ts : NULL, sigmask)) < 1) {
-		if (r == -1)
-			fido_log_error(errno, "%s: ppoll", __func__);
+	do {
+		r = ppoll(&pfd, 1, ms > -1 ? &ts : NULL, sigmask);
```

Comment:
Don't we have to account for the clock drift here as well if we are interrupted many times? Or why does this not matter in this case?

Reply:
Yes you are right. It seems that we even have to measure the elapsed time or rather calculate the remaining time our self due to the issue mentioned in the notes: https://linux.die.net/man/2/ppoll

Reply:
It is indeed quite complicated :smile: 

Doesn't this do the same?
```c
    if (clock_gettime(CLOCK_MONOTONIC, &ts_now) == -1) {
				fido_log_error(errno, "%s: clock_gettime", __func__);
				return (-1);
    }
    ts.tv_sec = ts_end.tv_sec - ts_now.tv_sec;
    ts.tv_nsec = ts_end.tv_nsec - ts_now.tv_nsec;
    if (ts.tvnsec < 0) {
      ts.tv_sec -= 1;
      ts.tv_nsec += 1000000000;
    }
    if (ts.tv_sec < 0) {
      return (-1); // Timeout expired
    }
```

Reply:
Simplified it